### PR TITLE
Add all Loans Models

### PR DESCRIPTION
**Pull request by troomtroom**

### CODE REVIEW

File: src/main/java/seedu/ichifund/model/loan/Loan.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
+package seedu.ichifund.model.loan;
+
+import static seedu.ichifund.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.ichifund.model.amount.Amount;
+
+import seedu.ichifund.model.date.Date;
+import seedu.ichifund.model.person.Name;
```
Comment (weiyang13) [inline src/main/java/seedu/ichifund/model/loan/Loan.java]: Hi, can you change this import to `loan.person` instead?

File: src/main/java/seedu/ichifund/model/loan/LoanDescription.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import static java.util.Objects.requireNonNull;
+import static seedu.ichifund.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Transaction's description in the transaction recorder.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLoanDescription(String)}
+ */
+public class LoanDescription implements Comparable<LoanDescription> {
+
+    public static final String MESSAGE_CONSTRAINTS =
```
Comment (weiyang13) [inline src/main/java/seedu/ichifund/model/loan/LoanDescription.java]: Is this different from `model.Description`?

Comment (troomtroom) [reply src/main/java/seedu/ichifund/model/loan/LoanDescription.java]: not too much, but the regex and stuff is different

File: src/main/java/seedu/ichifund/model/loan/Loan.java (hunk 1 above)
Comment (troomtroom) [reply src/main/java/seedu/ichifund/model/loan/Loan.java]: moved name to loan

**Comment by weiyang13** [comment]
@troomtroom there's some merge conflicts on your end. Please try to resolve them. Also, do confirm if you are still using `LoanDescription` instead of `Description` because I thought you mentioned otherwise.

**Pull request merged**